### PR TITLE
Feature/art registration flow

### PR DESCRIPTION
**Pull request by pjenys0**
Implements flow at https://pastel.wiki/en/Architecture/Workflows/NewArtRegistration
It includes all steps flow excepts:
- Generate Near duplicate fingerprints and Rareness and NSWF ranks from re-sampled image (img1-r)
- Generate “seen before” ranks from re-sampled image (img1-r)

Other note:
- go-webp is copied directly in repos. we will create a dedicated repo for it later after the Murata release.

Tickets:
https://freedcamp.com/notifications/panel/task/40128303
https://freedcamp.com/notifications/panel/task/40128253

### CODE REVIEW

**Comment by pjenys0** [comment]
@danileng20 , @matee81 : could you please help to review?

File: walletnode/services/artworkregister/task.go [hunk 1]
```diff
@@ -280,11 +372,392 @@ func (task *Task) pastelTopNodes(ctx context.Context) (node.List, error) {
...
+	src := bytes.NewReader(js)
+	if _, err := io.Copy(hasher, src); err != nil {
+		return nil, errors.Errorf("failed to hash identifiers file %w", err)
+	}
+	symbolIDFile.FileIdentifer = base58.Encode(hasher.Sum(nil))
+
+	return &symbolIDFile, nil
+}
+
+func (task *Task) createArtTicket(_ context.Context) error {
```
Comment (danileng20) [inline walletnode/services/artworkregister/task.go:484]: context is not used. why we don't remove it. except planing to use it in the future

File: walletnode/services/artworkregister/task.go [hunk 2]
```diff
@@ -280,11 +372,392 @@ func (task *Task) pastelTopNodes(ctx context.Context) (node.List, error) {
...
+	}
+
+	if err := task.nodes.MatchRegistrationFee(); err != nil {
+		return errors.Errorf("registration fees don't matched %w", err)
+	}
+
+	// check if fee is over-expection
+	task.registrationFee = task.nodes.RegistrationFee()
+	if task.registrationFee > int64(task.Request.MaximumFee) {
+		errors.Errorf("fee too high: registration fee %d, maximum fee %d", task.registrationFee, int64(task.Request.MaximumFee))
```
Comment (danileng20) [inline walletnode/services/artworkregister/task.go:733]: is return keyword missing?

File: pastel/go.mod [hunk 3]
```diff
@@ -5,7 +5,12 @@ go 1.16
...
 	github.com/pastelnetwork/gonode/common v0.0.0
+	github.com/stretchr/objx v0.2.0 // indirect
 	github.com/stretchr/testify v1.7.0
+	golang.org/x/sys v0.0.0-20210426230700-d19ff857e887 // indirect
 )
 
-replace github.com/pastelnetwork/gonode/common => ../common
+replace (
+	github.com/pastelnetwork/gonode/common => ../common
+	github.com/pastelnetwork/gonode/probe => ../probe
```
Comment (danileng20) [inline pastel/go.mod:15]: seem that it is not used

File: walletnode/services/artworkregister/task.go [hunk 4]
```diff
@@ -53,136 +97,182 @@ func (task *Task) run(ctx context.Context) error {
...
+		case <-ctx.Done():
+			return errors.Errorf("context done %w", ctx.Err())
+		case <-time.After(15 * time.Second):
+			checkConfirms := func() error {
+				subCtx, cancel := context.WithTimeout(ctx, 15*time.Second)
+				defer cancel()
+
+				result, err := task.pastelClient.GetRawTransactionVerbose1(subCtx, txID)
+				if err != nil {
+					errors.Errorf("failed to get transaction %w", err)
```
Comment (mateeullahmalik) [inline walletnode/services/artworkregister/task.go:200]: return?

File: walletnode/services/artworkregister/task.go (hunk 2 above)
Comment (leonidcode20) [reply walletnode/services/artworkregister/task.go]: Fixed.

File: pastel/go.mod (hunk 3 above)
Comment (leonidcode20) [reply pastel/go.mod:15]: probe is not used any more. Fixed

File: walletnode/services/artworkregister/task.go (hunk 4 above)
Comment (leonidcode20) [reply walletnode/services/artworkregister/task.go]: Fixed

File: walletnode/services/artworkregister/task.go (hunk 1 above)
Comment (leonidcode20) [reply walletnode/services/artworkregister/task.go:492]: Yes, it was planed for future use.

File: pastel/reg_ticket.go [hunk 5]
```diff
@@ -47,12 +47,29 @@ type ArtTicket struct {
 	AppTicketData AppTicket `json:"-"`
 }
 
+// AlternativeNSFWScore represents the not-safe-for-work of an image
+type AlternativeNSFWScore struct {
```
Comment (danileng20) [inline pastel/reg_ticket.go:51]: Why do we need this duplicated struct?

File: pastel/reg_ticket.go [hunk 6]
```diff
@@ -47,12 +47,29 @@ type ArtTicket struct {
...
+
 // FingerAndScores represents structure of app ticket
 type FingerAndScores struct {
-	FingerprintData []byte
-	RarenessScore   int
-	NSFWScore       int
-	SeenScore       int
+	DupeDectectionSystemVersion string               `json:"dupe_dectection_system_version"`
+	HashOfCandidateImageFile    string               `json:"hash_of_candidate_image_file"`
+	IsLikelyDupe                int                  `json:"is_likely_dupe"`
```
Comment (danileng20) [inline pastel/reg_ticket.go:63]: As confirm, we will remove this field.

**Review by danileng20 — DISMISSED** [review]
LGTM! Except for some duplicate code, we can optimize later.

**Review by leonidcode20 — APPROVED** [review]

**Review by leonidcode20 — APPROVED** [review]

**Review by georgkcode0 — APPROVED** [review]
LGTM

**Pull request merged**